lib/npm.js

```diff
@@ -55,16 +56,21 @@ class npm {
   }
 
   isAuthenticated() {
-    return this.shell.run('npm whoami').then(() => true, () => false);
+    const registry = this.getRegistry(this.options);
+    return this.shell.run(`npm whoami --registry ${registry}`).then(() => true, () => false);
```

Comment:
At this point we could omit the flag if it's the default registry. I'm always appending it for simpler code, but ofc something we could change if you want that.